examples/main_test.go

```diff
@@ -0,0 +1,33 @@
+package examples
+
+import (
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+)
+
+func TestMain(m *testing.M) {
+	// Ensure we use a locally built provider
+	awsProviderPath, err := exec.LookPath("pulumi-resource-aws")
+	if err != nil {
+		log.Println("AWS provider not found in path!")
```

Comment:
What if you auto-correct this problem? For example you do os.SetEnv to change the PATH, would it work?